cachito/workers/scm.py

```diff
@@ -147,7 +171,30 @@ def fetch_source(self):
             ) as exc:
                 log.warning("Error handling archived artifact '%s': %s", previous_archive, exc)
 
-        self.clone_and_archive()
+        self.clone_and_archive(gitsubmodule)
+
+    def process_git_submodule(self, submodule):
+        """Clone the repo for the git submodule.
+
+        Clone git submodule repo for the Cachito request under appropriate directory. For Cachito
+        request for `retrodep` directory and `go-github` submodule, the dir structure would
+        look like, retrodep/go-github/<content_of_go-github_repo>
+
+        :param git.Repo.submodule submodule: Git submodule for the Cachito request repo
+        """
+        try:
+            repo = git.repo.Repo.clone_from(
+                submodule.url,
+                submodule.abspath,
+                no_checkout=True,
+                # Don't allow git to prompt for a username if we don't have access
+                env={"GIT_TERMINAL_PROMPT": "0"},
+            )
+
+            self._reset_git_head(repo, submodule.hexsha)
```

Comment:
If I'm not mistaken, you could replace all of this with `submodule.update()`

Reply:
This works. How come I didn't come across that option. Where did you find it? -_-

Reply:
I found it here: https://gitpython.readthedocs.io/en/stable/tutorial.html#submodule-handling

 But honestly, the only reason I knew `update()` would do what you need is that I have used submodules before and I was confused why the command used for fetching was `git submodule update` :D